cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1178,11 +1180,8 @@ private void deleteApp(ApplicationId appId, ApplicationSpecification spec) throw
     deleteAppMetadata(appId, spec);
     store.deleteWorkflowStats(appId);
     deletePreferences(appId, spec);
-
-    // version specific deletion
-    for (ApplicationId versionAppId : store.getAllAppVersionsAppIds(appId)) {
```

Comment:
I think we weren't fully deleting all versions of an app... like metrics and metadata, instead we should call ` deleteApp(ApplicationId appId, ApplicationSpecification spec)` from upstream this method when we delete an appreference